smartconnect/__init__.py

```diff
@@ -178,6 +198,80 @@ def get_conservation_areas(self) -> List[models.ConservationArea]:
+
+        config_datamodels = requests.get(f'{self.api}/api/metadata/configurablemodel',
+                                        auth=self.auth,
+                                        params={"ca_uuid": ca_uuid},
+                                        headers={
+                                            'accept': 'application/json',
+                                        },
+                                        stream=True,
+                                        verify=self.verify_ssl,
+                                        timeout=DEFAULT_TIMEOUT)
+
+        config_datamodels.raw.decode_content = True
+
+        self.logger.info(f'Get configurable data models for CA {ca_uuid} data model took {config_datamodels.elapsed.total_seconds()} seconds',
+                         extra=dict(**extra_dict,
+                                    status_code=config_datamodels.status_code))
+
+        if not config_datamodels.ok:
+            self.logger.error(
+                f'Failed to download configurable datamodels for  CA {ca_uuid}. Status_code is: {config_datamodels.status_code}',
+                extra=dict(**extra_dict,
+                           status_code=config_datamodels.status_code))
+            raise Exception('Failed to download Configurable Data Models')
+
+        cdms = config_datamodels.json()
+        er_cdm = next((x for x in cdms if x.get('name') == 'James Test Model'), None)
+        if er_cdm:
+            return er_cdm.get('uuid')
+        else:
+            return None
+
+    def download_configurable_datamodel(self, *, cm_uuid: str = None):
+        extra_dict = dict(cm_uuid=cm_uuid,
+                          url=f'{self.api}/api/metadata/configurablemodel/{cm_uuid}')
+
+        # config_datamodel = open('chunga-config-datamodel-response.xml', 'r')
+
+        config_datamodel = requests.get(f'{self.api}/api/metadata/configurablemodel/{cm_uuid}',
+                                    auth=self.auth,
+                                    headers={
+                                        'accept': 'application/xml',
+                                    },
+                                    stream=True,
+                                    verify=self.verify_ssl,
+                                    timeout=DEFAULT_TIMEOUT)
+        config_datamodel.raw.decode_content = True
+
+        self.logger.info(f'Download configurable model {cm_uuid} data model took {config_datamodel.elapsed.total_seconds()} seconds',
+                         extra=dict(**extra_dict,
+                                    status_code=config_datamodel.status_code))
+
+        if not config_datamodel.ok:
+            self.logger.error(
+                f'Failed to download data model for  configurable model {cm_uuid}. Status_code is: {config_datamodel.status_code}',
+                extra=dict(**extra_dict,
+                           status_code=config_datamodel.status_code))
+            raise Exception('Failed to download Data Model')
```

Comment:
Sounds good. I created https://allenai.atlassian.net/browse/SIK-1635 to capture evaluating this and including it in the set of next changes. Going to get this merged now so we have latest in Dev since I wont have capacity to work on it for some time